src/shared/ui/components/widget-tab/widget-tab.component.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useMemo } from 'react';
+
+import { useWidgetTabs } from '../../providers/widget-tabs';
+import { WidgetBase } from '../widget-base';
+import { classes } from '../../utils/classes';
+
+import { TabHandle } from './tab-handle.component';
+import { WidgetTabProperties } from './widget-tab.types';
+
+export const WidgetTab = ({
+  top,
+  theme,
+  twitterHandle,
+  tabImage,
+  tabName,
+  children,
+  className,
+  closeButtonClassName,
+  onClose,
+}: WidgetTabProperties) => {
+  const {
+    tabs,
+    preferredTab,
+    addWidgetTab,
+    setUserPreferredTab,
+    removeAllUserWidgets,
+  } = useWidgetTabs();
+
+  const userTabs = useMemo(() => {
+    return tabs[twitterHandle];
+  }, [tabs, twitterHandle]);
+
+  const userPreferredTab = preferredTab[twitterHandle] ?? userTabs?.at(0);
+
+  const left = useMemo(() => {
+    const index = userTabs?.indexOf(tabName) ?? -1;
+    if (index !== -1) {
+      return index * 100;
+    }
+
+    return 0;
+  }, [tabName, userTabs]);
+
+  useEffect(() => {
+    addWidgetTab(twitterHandle, tabName);
+  }, [addWidgetTab, tabName, twitterHandle]);
```

Comment:
is it okay if this effect will re-run multiple times in a lifecycle of single WidgetTab? I am wondering if it would be better if we used `useEffectOnce` coming from `react-use` library which we have installed, but I might be completely wrong :think:

Reply:
oh, had no idea about that function. In theory, the dep params that are there should not change, but I'll use the `useEffectOnce`